File: src/main/java/org/profit/main/CommandLine.java
package org.profit.main;

import org.profit.ProfitServer;
import org.profit.config.spring.AppSpringConfig;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

/**
 * @author Kyia
 */
public class CommandLine {

	protected CommandLine() {

	}

	public static void main(String args[]) {
		CommandLine cli = new CommandLine();
		try {
			// Get configuration
			ProfitServer server = cli.getConfiguration(args);
			if (server == null) {
				return;
			}

			// Start the server
			server.start();

			cli.addShutdownHook(server);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	private void addShutdownHook(final ProfitServer server) {
		Runnable shutdownHook = new Runnable() {
			@Override
			public void run() {
				server.stop();
			}
		};

		// Add shutdown hook
		Runtime runtime = Runtime.getRuntime();
		runtime.addShutdownHook(new Thread(shutdownHook));
	}

	protected ProfitServer getConfiguration(String args[]) throws Exception {
		ProfitServer server = null;
		if (args.length == 0) {
			System.out.println("Using default configuration.");

			AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(AppSpringConfig.class);
			//ctx.registerShutdownHook();

			server = new ProfitServer();
		}
		return server;
	}
}

File: src/main/java/org/profit/main/CommandLine.kt
package org.profit.main

import org.profit.ProfitServer
import org.profit.config.spring.AppSpringConfig
import org.springframework.context.annotation.AnnotationConfigApplicationContext

class CommandLine protected constructor() {
    private fun addShutdownHook(server: ProfitServer) {
        val shutdownHook = Runnable { server.stop() }
        // Add shutdown hook
        val runtime = Runtime.getRuntime()
        runtime.addShutdownHook(Thread(shutdownHook))
    }

    @Throws(Exception::class)
    protected fun getConfiguration(args: Array<String>): ProfitServer? {
        var server: ProfitServer? = null
        if (args.size == 0) {
            println("Using default configuration.")
            val ctx = AnnotationConfigApplicationContext(AppSpringConfig::class.java)
            //ctx.registerShutdownHook();
            server = ProfitServer()
        }
        return server
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val cli = CommandLine()
            try { // Get configuration
                val server = cli.getConfiguration(args) ?: return
                // Start the server
                server.start()
                cli.addShutdownHook(server)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }
}